smartsim/_core/launcher/dragon/dragonLauncher.py

```diff
@@ -214,7 +221,23 @@ def connect_to_dragon(self, path: str) -> None:
 
                 launcher_socket.close()
                 self._set_timeout(self._timeout)
-                self._handsake(dragon_head_address)
+                self._handshake(dragon_head_address)
+
+                def cleanup() -> None:
+                    try:
+                        with self._comm_lock:
+                            shutdown_req = DragonShutdownRequest()
+                            self._send_request_as_json(shutdown_req)
+                    except zmq.error.ZMQError as e:
+                        logger.error(
+                            "Could not send shutdown request to dragon server,"
+                            f" ZMQ error: {e}"
+                        )
+                    finally:
+                        if self._dragon_head_process is not None:
+                            os.kill(self._dragon_head_process.pid, signal.SIGINT)
```

Comment:
OK, this was only half-done. I moved everything to a static method, as [this post](https://stackoverflow.com/questions/16333054/what-are-the-implications-of-registering-an-instance-method-with-atexit-in-pytho) suggests. Can you give it a look?

Reply:
Totally agree with the sentiment to keep the first `DragonLauncher` instance out of memory for the entire length of the program! This change seems to make sense to me! My only suggestion would be to perhaps mark `dragon_cleanup` as "protected" (e.g. `_dragon_cleanup`) so that it is not used outside of this module!

Reply:
Done